packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/gollm/GoLLMController.java

```diff
@@ -440,6 +441,111 @@ public ResponseEntity<TaskResponse> createConfigureModelFromDatasetTask(
 		return ResponseEntity.ok().body(resp);
 	}
 
+	@GetMapping("/interventions-from-document")
+	@Secured(Roles.USER)
+	@Operation(summary = "Dispatch a `GoLLM interventions-from-document` task")
+	@ApiResponses(
+		value = {
+			@ApiResponse(
+				responseCode = "200",
+				description = "Dispatched successfully",
+				content = @Content(
+					mediaType = "application/json",
+					schema = @io.swagger.v3.oas.annotations.media.Schema(implementation = TaskResponse.class)
+				)
+			),
+			@ApiResponse(
+				responseCode = "404",
+				description = "The provided model or document arguments are not found",
+				content = @Content
+			),
+			@ApiResponse(responseCode = "500", description = "There was an issue dispatching the request", content = @Content)
+		}
+	)
+	public ResponseEntity<TaskResponse> createInterventionsFromDocumentTask(
+		@RequestParam(name = "model-id", required = true) final UUID modelId,
+		@RequestParam(name = "document-id", required = true) final UUID documentId,
+		@RequestParam(name = "mode", required = false, defaultValue = "ASYNC") final TaskMode mode,
+		@RequestParam(name = "workflow-id", required = false) final UUID workflowId,
+		@RequestParam(name = "node-id", required = false) final UUID nodeId,
+		@RequestParam(name = "project-id", required = false) final UUID projectId
+	) {
+		final Schema.Permission permission = projectService.checkPermissionCanRead(
+			currentUserService.get().getId(),
+			projectId
+		);
+
+		// Grab the document
+		final Optional<DocumentAsset> document = documentAssetService.getAsset(documentId, permission);
+		if (document.isEmpty()) {
+			log.warn(String.format("Document %s not found", documentId));
+			throw new ResponseStatusException(HttpStatus.NOT_FOUND, messages.get("document.not-found"));
+		}
```

Comment:
@dvince2 all those `.getAsset()` that returns Optional, should they throw the error below instead?
I fill that the code should be simpler in this controller, and easy to mess around.


Don't mind me, this is not worth of a refactor, just thinking about it, and I would like your opinion about it.